File: server-scala-play-server/app/model/SubscriptionPlanResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for SubscriptionPlanResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class SubscriptionPlanResponse(
  id: Option[Long],
  name: Option[String],
  description: Option[String],
  options: Option[List[SubscriptionOptionResponse]],
  price: Option[Double],
  promo: Option[Double],
  transactionFee: Option[Double],
  role: Option[String],
  defaultPlan: Option[Boolean],
  visible: Option[Boolean],
  image: Option[AssetShortResponse]
)

object SubscriptionPlanResponse {
  implicit lazy val subscriptionPlanResponseJsonFormat: Format[SubscriptionPlanResponse] = Json.format[SubscriptionPlanResponse]
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/SubscriptionPlanResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class SubscriptionPlanResponse 
{
    public long Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public List<SubscriptionOptionResponse> Options { get; set; }
    public double Price { get; set; }
    public double Promo { get; set; }
    public double TransactionFee { get; set; }
    public string Role { get; set; }
    public bool DefaultPlan { get; set; }
    public bool Visible { get; set; }
    public AssetShortResponse Image { get; set; }
}


